lib/ansible/modules/cloud/vmware/vcenter_plugin.py

```diff
@@ -0,0 +1,225 @@
+version_added: 2.8
+author:
+    - Michael Tipton (@castawayegr)
+notes:
+    - Tested on vSphere 6.5
+requirements:
+    - "python >= 2.6"
+    - PyVmomi
+options:
+  extension_key:
+    description:
+    - The extension key of the plugin to install or uninstall.
+    required: True
+  version:
+    description:
+    - The version of the plugin you are installing or uninstalling.
+    required: True
+  name:
+    description:
+    - Required for C(state=present). The name of the plugin you are installing.
+  company:
+    description:
+    - Required for C(state=present). The name of the company that makes the plugin.
+  description:
+    description:
+    - Required for C(state=present). A short description of the plugin.
+  email:
+    description:
+    - Required for C(state=present). Administrator Email to use for Plugin.
+  url:
+    description:
+    - Required for C(state=present). Link to server hosting plugin zip file to install.
+  ssl_thumbprint:
+    description:
+    - Required for C(state=present). SSL thumbprint of the plugin hosting server.
+  server_type:
+    description:
+    - Required for C(state=present). Type of server being used to install the plugin (SOAP, REST, HTTP, etc.).
+    default: vsphere-client-serenity
+  client_type:
+    description:
+    - Required for C(state=present). Type of client the plugin is (win32, .net, linux, etc.).
+    default: vsphere-client-serenity
+  visible:
+    description:
+    - Show the plugin in solution manager inside vCenter.
+    default: True
+    type: bool
+  state:
+    description:
+    - Add or remove vCenter Plugin.
+    choices: [absent, present]
+    default: present
+extends_documentation_fragment: vmware.documentation
+'''
+
+EXAMPLES = '''
+    - name: Register vCenter Plugin
+      vcenter_plugin:
+         hostname: "{{ groups['vcsa'][0] }}"
+         username: "{{ vcenter_username }}"
+         password: "{{ site_password }}"
+         extension_key: "{{ extension_key }}"
+         version: "1.0"
+         company: "Acme"
+         name: "Acme Plugin"
+         description: "acme management"
+         email: "user@example.com"
+         url: "https://10.0.0.1/ACME-vSphere-web-plugin-1.0.zip"
+         ssl_thumbprint: "{{ ssl_thumbprint }}"
+         state: present
+      delegate_to: localhost
+      register: register_plugin
+
+    - name: Deregister vCenter Plugin
+      vcenter_plugin:
+         hostname: "{{ groups['vcsa'][0] }}"
+         username: "{{ vcenter_username }}"
+         password: "{{ site_password }}"
+         extension_key: "{{ extension_key }}"
+         version: "1.0"
+         state: absent
+      delegate_to: localhost
+      register: deregister_plugin
+'''
+
+RETURN = """
+result:
+    description: information about performed operation
+    returned: always
+    type: string
+    sample: "'com.acme.plugin' installed."
+"""
+
+try:
+    from pyVmomi import vim, vmodl
+except ImportError:
+    pass
+
+try:
+    import datetime
```

Comment:
I think datetime is basic datatype of Python and is available in all installation. So this check will not be required.